### PR TITLE
Fix repodata for old gcc_*

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
cc @beckermr

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -628,6 +628,19 @@ def _gen_new_index(repodata, subdir):
...
+            subdir in ["linux-64", "linux-aarch64", "linux-ppc64le"]
+            and record_name in [
+                "gcc_impl_" + subdir, "gxx_impl_" + subdir, "gfortran_impl_" + subdir]
+            and record['version'] in ['5.4.0', '7.2.0', '7.3.0', '8.2.0']
+        ):
+            new_constrains = record.get('constrains', [])
+            for pkg in ["libgcc-ng", "libstdcxx-ng", "libgfortran", "libgomp"]:
+                new_constrains.append("{} 5.4.*|7.2.*|7.3.*|8.2.*|9.1.*|9.2.*".format(pkg))
+            new_constrains.append("binutils_impl_" + subdir + " <2.34")
+            new_constrains.append("ld_impl_" + subdir + " <2.34")
```
Comment (beckermr) [inline recipe/gen_patch_json.py:641]: Why these?

Comment (isuruf) [reply recipe/gen_patch_json.py:641]: It's the same as https://github.com/conda-forge/binutils-feedstock/blob/master/recipe/meta.yaml#L42-L44, but I'd like to remove the constraints from the recipe there and have it here.

Comment (beckermr) [reply recipe/gen_patch_json.py:641]: ahhhhh thank you

**Pull request merged**